src/main/java/ch/njol/skript/effects/Delay.java

```diff
@@ -41,100 +33,106 @@
-@Examples({"wait 2 minutes",
-		"halt for 5 minecraft hours",
-		"wait a tick"})
+@Examples({
+	"wait 2 minutes",
+	"halt for 5 minecraft hours",
+	"wait a tick"
+})
 @Since("1.4")
 public class Delay extends Effect {
+
 	static {
 		Skript.registerEffect(Delay.class, "(wait|halt) [for] %timespan%");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	protected Expression<Timespan> duration;
 
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		getParser().setHasDelayBefore(Kleenean.TRUE);
+
 		duration = (Expression<Timespan>) exprs[0];
 		if (duration instanceof Literal) { // If we can, do sanity check for delays
 			long millis = ((Literal<Timespan>) duration).getSingle().getMilliSeconds();
 			if (millis < 50) {
 				Skript.warning("Delays less than one tick are not possible, defaulting to one tick.");
 			}
 		}
+
 		return true;
 	}
 
 	@Override
 	@Nullable
-	protected TriggerItem walk(final Event e) {
+	protected TriggerItem walk(Event e) {
 		debug(e, true);
-		final long start = Skript.debug() ? System.nanoTime() : 0;
-		final TriggerItem next = getNext();
+		long start = Skript.debug() ? System.nanoTime() : 0;
+		TriggerItem next = getNext();
 		if (next != null && Skript.getInstance().isEnabled()) { // See https://github.com/SkriptLang/Skript/issues/3702
 			delayed.add(e);
-			final Timespan d = duration.getSingle(e);
-			if (d == null)
+
+			Timespan duration = this.duration.getSingle(e);
+			if (duration == null)
 				return null;
 			
 			// Back up local variables
 			Object localVars = Variables.removeLocals(e);
 			
-			Bukkit.getScheduler().scheduleSyncDelayedTask(Skript.getInstance(), new Runnable() {
-				@Override
-				public void run() {
-					if (Skript.debug())
-						Skript.info(getIndentation() + "... continuing after " + (System.nanoTime() - start) / 1000000000. + "s");
-					
-					// Re-set local variables
-					if (localVars != null)
-						Variables.setLocalVariables(e, localVars);
-					
-					Object timing = null;
-					if (SkriptTimings.enabled()) { // getTrigger call is not free, do it only if we must
-						Trigger trigger = getTrigger();
-						if (trigger != null) {
-							timing = SkriptTimings.start(trigger.getDebugLabel());
-						}
+			Bukkit.getScheduler().scheduleSyncDelayedTask(Skript.getInstance(), () -> {
+				if (Skript.debug())
+					Skript.info(getIndentation() + "... continuing after " + (System.nanoTime() - start) / 1000000000. + "s");
```

Comment:
```suggestion
					Skript.info(getIndentation() + "... continuing after " + (System.nanoTime() - start) / 1_000_000_000. + "s");
```
to be a bit more clear